### ISSUE TITLE
Torrent Straming

### ISSUE THREAD

**Issue by Kosava** [xylosper/bomi#228]
It would be nice to have it, I know you mentioned on BBS, you dont know how to implement it, but let it be feature request until you figure.

**Comment by ValdikSS** [comment]
https://github.com/ValdikSS/aceproxy

**Comment by Rabcor** [comment]
This would be a cool feature to have, I hear it's all the rage these days among my friends, I've never tried it myself though, but it sounds pretty cool to me.

**Comment by ahjolinna** [comment]
I have no idea how popcorn time does it...btw. it would be nice to use bomi as 3rd-party videoplayer (only option is now mplayer and vlc)

I have no idea if this helps but I found this also: 
https://github.com/mafintosh/peerflix
https://github.com/mafintosh/torrent-stream

**Comment by xylosper** [comment]
@ahjolinna 
All script based solutions are out of scope because they will bring unpleasant additional dependencies.
Now bomi aims cross-platform(Linux/Windows) and it's not easy to provide such dependency on non Linux OS unless upstream provides a precompiled standalone executable like youtube-dl.
Futhermore, I would never use peerflix because it just produces lots of leeches in practice.

For now, I'm looking into torrent-stream source code to try to port it to QML or C++.

**Comment by ValdikSS** [comment]
>For now, I'm looking into torrent-stream source code to try to port it to QML or C++.

Please don't. That old decompiled version contains a lot of bugs and the acestream overall project is driven by literally mentally ill people which won't answer you if you write in English or would speak rude to you if you talk to them in Russian.

**Comment by xylosper** [comment]
> That old decompiled version contains a lot of bugs and the acestream overall project is driven by literally mentally ill people

I'm not sure what you're talking about. What is acestream overall project?
Is the developer of torrent-stream from Russian?
As far as I can see from his profile, I think he came from Denmark.

**Comment by ValdikSS** [comment]
Oh, I see, you're talking about https://github.com/mafintosh/torrent-stream. There is another project called AceStream, which was called TorrentStream a while ago.

**Comment by kehugter** [comment]
Non-technical question: wouldn't torrent streaming put bomi's github repo at risk, given all the drama that such applications have generated in the last 2 years?

**Comment by xylosper** [comment]
I'm not sure what you mean. If you mean some kind of service like popcorntime, it won't happen for legal issue.

**Comment by ahjolinna** [comment]
it seems SMPlayer is now supporting this, and they are using peerflix

http://blog.smplayer.info/how-to-play-torrents-in-smplayer/

**Comment by ValdikSS** [comment]
No it doesn't. It's just peerflix which generates streaming link.

**Comment by xylosper** [comment]
@ahjolinna 
As I already told, peerfilx is not a good solution because peerflix makes all clients leech although integration of peerflix is obviously easiest solution.
